loadbalancer/messages/messages.ipldsch

```diff
@@ -0,0 +1,46 @@
+type RequestKind enum {
+	| New ("n")
+  | Extend ("e")
+  | Terminate ("t")
+} representation string
+
+type ProtocolID string
+type RoutingRequest struct {
+   Kind RequestKind (rename "k")
+   Protocols [ProtocolID] (rename "ps")
+} representation map
+
+type ResponseCode enum {
+  | Ok ("o")
+  | Rejected ("r")
+} representation string
+
+type DateTime int
+
+type RoutingResponse struct {
+  Code ResponseCode (rename "c")
+  Message String (rename "m") # more info if rejected
+  Expiry optional DateTime
+} representation map
+
+type ForwardingKind enum {
+  | Inbound ("i")
+  | Outbound ("o")
+} representation string
+
+type PeerID string
+type PubKey bytes
+
+type ForwardingRequest struct {
+   Kind ForwardingKind (rename "k")
```

Comment:
another case for a union here; we should look at doing this so we have protocol level enforcement of types

Reply:
Indeed. At the moment, the bindnode go union type implementation sucks enough that I don't want to deal with it. (I believe currently it's a struct with a bunch of nullable members?)